src/reward_splitter/tasks.py

```diff
@@ -0,0 +1,243 @@
+    MULTICALL_BATCH_SIZE,
+    REWARD_SPLITTER_INTERVAL,
+    REWARD_SPLITTER_MIN_ASSETS,
+    settings,
+)
+from src.reward_splitter.contracts import RewardSplitterContract, RewardSplitterEncoder
+from src.reward_splitter.graph import (
+    graph_get_claimable_exit_requests,
+    graph_get_reward_splitters,
+)
+from src.reward_splitter.typings import ExitRequest, RewardSplitter
+
+logger = logging.getLogger(__name__)
+
+
+class SplitRewardTask(BaseTask):
+
+    # pylint: disable-next=too-many-locals
+    async def process_block(self, interrupt_handler: InterruptHandler) -> None:
+        """
+        Processes reward splitters for the vaults specified in settings.
+
+        This function performs the following steps:
+        - Retrieves reward splitters associated with the vaults from Subgraph.
+        - Retrieves claimable exit requests for the reward splitters.
+        - Generates multicall contract calls for each reward splitter.
+        - Processes the multicall batches and waits for transaction confirmations.
+        """
+        block = await execution_client.eth.get_block('finalized')
+
+        app_state = AppState()
+        if not await _check_reward_splitter_block(app_state, block['number']):
+            return
+
+        # check current gas prices
+        gas_manager = build_gas_manager()
+        if not await gas_manager.check_gas_price():
+            return
+
+        logger.info('Fetching reward splitters')
+        reward_splitters = await graph_get_reward_splitters(
+            block_number=block['number'], vaults=settings.vaults
+        )
+
+        if not reward_splitters:
+            logger.info('No reward splitters found for given vaults')
+            return
+
+        splitter_to_exit_requests = await graph_get_claimable_exit_requests(
+            block_number=block['number'], receivers=[rs.address for rs in reward_splitters]
+        )
+
+        # Multicall contract calls per address
+        calls: dict[ChecksumAddress, list[HexStr]] = {}
+
+        for reward_splitter in reward_splitters:
+            logger.info(
+                'Processing reward splitter %s for vault %s',
+                reward_splitter.address,
+                reward_splitter.vault,
+            )
+            vault = reward_splitter.vault
+
+            harvest_params = await get_harvest_params(vault)
+            exit_requests = splitter_to_exit_requests.get(reward_splitter.address, [])  # nosec
+
+            reward_splitter_calls = await _get_reward_splitter_calls(
+                reward_splitter=reward_splitter,
+                harvest_params=harvest_params,
+                exit_requests=exit_requests,
+            )
+
+            calls[reward_splitter.address] = reward_splitter_calls
+        if not calls:
+            logger.warning('No calls to process')
+            return
+
+        calls_batches = []
+        calls_batch: list[tuple[ChecksumAddress, HexStr]] = []
+        for address, address_calls in calls.items():
```

Comment:
You cannot use multicall contract here as the permission check in the RewardSplitter will fail. The caller must be `claimer`.